File: 4.Programming course/Python language/readgpio.py
import wiringpi

GPIO_Output_Pin = 0
GPIO_Intput_Pin = 1
OUTPUT = 1
INPUT = 0
HIGH = 1
LOW = 0

wiringpi.wiringPiSetup()
wiringpi.pinMode(GPIO_Output_Pin,OUTPUT)
wiringpi.pinMode(GPIO_Intput_Pin,INPUT)

while 1:
    wiringpi.digitalWrite(GPIO_Output_Pin,HIGH)
    print('GPIO_Output_Pin OUTPUT =>HIGH')
    print('GPIO_Intput_Pin INPUT <=',wiringpi.digitalRead(GPIO_Intput_Pin))
    wiringpi.delay(1000)
    print()

    wiringpi.digitalWrite(GPIO_Output_Pin,LOW)
    print('GPIO_Output_Pin OUTPUT =>LOW')
    print('GPIO_Intput_Pin INPUT <=',wiringpi.digitalRead(GPIO_Intput_Pin))
    wiringpi.delay(1000)
    print()

File: 4.Programming course/C language/read_gpio.c
#include<stdio.h>
#include<wiringPi.h>

#define Output_Pin 0
#define Input_Pin 1
int main()
{
	printf("This is an example of a raspberry PI pin reading\n");
	wiringPiSetup();
	pinMode(Output_Pin,OUTPUT);
	pinMode(Input_Pin,INPUT);
	
	printf("Set Output_Pin : H\n");
    digitalWrite(Output_Pin,HIGH);
    delay(200);
	printf("Input_Pin : %d\n",digitalRead(Input_Pin));
    delay(2000);
    
	printf("Set Output_Pin: L\n");
	digitalWrite(Output_Pin,LOW);
	delay(200);
	printf("Input_Pin : %d\n",digitalRead(Input_Pin));
	while(1)
	{

	}
	return 0;
}

